src/modules/library/document_models.py

```diff
@@ -0,0 +1,199 @@
+
+        Args:
+            text (str): text, form which we take words and compute their average embedding
+
+        Returns:
+            (numpy.ndarray): an average of word embeddings of words that appear in
+            given text.
+        """
+
+        embedding = np.zeros(self.word_vectors.vector_size, dtype=np.float32)
+        if text is None:
+            return embedding
+
+        word_sorted = self.tokenize(text)
+        norm = 0
+        for token, number_of_appearances in word_sorted:
+            # sum all tokens embeddings of the vector
+            if token in self.word_vectors.vocab.keys():
+                embedding += self.word_vectors[token]
+                norm += 1
+
+        # return the normalized embedding; if not zero
+        return embedding if norm == 0 else embedding / norm
+
+    def embed_list_of_documents(self, docs):
+        """
+        Returns a document embedding of documents given as an argument.
+
+        Args:
+            docs (list(string)): list of documents we want to embed
+
+        Returns:
+             numpy.ndarray: matrix of document embeddings, where i-th line is the embedding of i-th document of the
+                 list 'docs'.
+        """
+
+        embedding = np.zeros((len(docs), self.word_vectors.vector_size), dtype=np.float32)
+        # embed individual documents:
+        for id, document in enumerate(docs):
+            embedding[id,:] = self.average_word_embedding(document)
+        return embedding
+
+    def embed_documents(self):
+        """
+        Creates the dataset embeddings for documents given at initialization and saves it in the attribute 'embedding'.
+        """
+
+        self.embedding = self.embed_list_of_documents(self.documents)
+
+    def add_documents(self, docs):
+        """
+        Appends given documents to the model's attribute 'documents' and adds lines for document embeddings of those
+        documents at the end of the matrix 'embedding'.
+
+        Args:
+            docs (list(str)): List of documents we want to add to our model
+        """
+
+        docs = [d for d in docs if d not in self.documents]
+        new_embedding = self.embed_list_of_documents(docs)
+        self.documents = self.documents + docs
+        self.embedding = np.concatenate((self.embedding, new_embedding))
+
+    def remove_documents(self, docs):
+        """
+        Removes given documents from the model's attribute 'documents' and removes lines for their embeddings from the
+        matrix 'embedding'.
+
+        Args:
+            docs (list(str)): List of documents we want to remove from our model
+        """
+
+        for doc in docs:
+            if doc in self.documents:
+                index = self.documents.index(doc)
+                self.documents.remove(doc)
+                self.embedding = np.concatenate((self.embedding[0:index, :], self.embedding[index+1:, :]))
```

Comment:
Return the number of documents that were removed from the model - a simple notification of what was done